deployment/ccip/changeset/cs_ccip_home.go

```diff
@@ -25,80 +35,150 @@ var (
+			DeltaGrace:                              internal.DeltaGrace,
+			DeltaCertifiedCommitRequest:             internal.DeltaCertifiedCommitRequest,
+			DeltaStage:                              internal.DeltaStage,
+			Rmax:                                    internal.Rmax,
+			MaxDurationQuery:                        internal.MaxDurationQuery,
+			MaxDurationObservation:                  internal.MaxDurationObservation,
+			MaxDurationShouldAcceptAttestedReport:   internal.MaxDurationShouldAcceptAttestedReport,
+			MaxDurationShouldTransmitAcceptedReport: internal.MaxDurationShouldTransmitAcceptedReport,
+		},
+		ExecuteOffChainConfig: pluginconfig.ExecuteOffchainConfig{
+			BatchGasLimit:             internal.BatchGasLimit,
+			RelativeBoostPerWaitHour:  internal.RelativeBoostPerWaitHour,
+			InflightCacheExpiry:       *config.MustNewDuration(internal.InflightCacheExpiry),
+			RootSnoozeTime:            *config.MustNewDuration(internal.RootSnoozeTime),
+			MessageVisibilityInterval: *config.MustNewDuration(internal.FirstBlockAge),
+			BatchingStrategyID:        internal.BatchingStrategyID,
+			TokenDataObservers:        tokenDataObservers,
+		},
+		CommitOffChainConfig: pluginconfig.CommitOffchainConfig{
+			RemoteGasPriceBatchWriteFrequency:  *config.MustNewDuration(internal.RemoteGasPriceBatchWriteFrequency),
+			TokenPriceBatchWriteFrequency:      *config.MustNewDuration(internal.TokenPriceBatchWriteFrequency),
+			TokenInfo:                          tokenInfo,
+			PriceFeedChainSelector:             ccipocr3.ChainSelector(feedChainSel),
+			NewMsgScanBatchSize:                merklemulti.MaxNumberTreeLeaves,
+			MaxReportTransmissionCheckAttempts: 5,
+			RMNEnabled:                         os.Getenv("ENABLE_RMN") == "true", // only enabled in manual test
+			RMNSignaturesTimeout:               30 * time.Minute,
+			MaxMerkleTreeSize:                  merklemulti.MaxNumberTreeLeaves,
+			SignObservationPrefix:              "chainlink ccip 1.6 rmn observation",
+		},
+	}
+}
+
 type PromoteAllCandidatesChangesetConfig struct {
 	HomeChainSelector uint64
 
-	// DONChainSelector is the chain selector of the DON that we want to promote the candidate config of.
+	// DONChainSelectors is the chain selector of the DONs that we want to promote the candidate config of.
 	// Note that each (chain, ccip capability version) pair has a unique DON ID.
-	DONChainSelector uint64
+	DONChainSelectors []uint64
 
 	// MCMS is optional MCMS configuration, if provided the changeset will generate an MCMS proposal.
 	// If nil, the changeset will execute the commands directly using the deployer key
 	// of the provided environment.
 	MCMS *MCMSConfig
 }
 
-func (p PromoteAllCandidatesChangesetConfig) Validate(e deployment.Environment, state CCIPOnChainState) (donID uint32, err error) {
+func (p PromoteAllCandidatesChangesetConfig) Validate(e deployment.Environment, state CCIPOnChainState) ([]uint32, error) {
```

Comment:
Should we include the validation - if mcmscfg is provided, validate the ownership is correctly set
Probably a common func would be nice to check whether ownership is set to MCMS